### PR TITLE
Default value support

### PR DESCRIPTION
**Pull request by ambiennt**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -2,6 +2,8 @@ cmake_minimum_required(VERSION 3.0)
 
 project(jsoncpp)
 
+set(CMAKE_CXX_STANDARD 20)
```
Comment (ZeroMemes) [inline CMakeLists.txt:5]: Unrelated to this PR. Rem

Comment (ZeroMemes) [inline CMakeLists.txt]: Unrelated to this PR, please revert.

**Review by ZeroMemes — CHANGES_REQUESTED** [review]
Replace exceptions with return default to match behavior of mcbe implementation

File: CMakeLists.txt (hunk 1 above)
Comment (ambiennt) [reply CMakeLists.txt]: If the standard isn't at least 17, MSVC selects 14 as a default and emits a warning saying that [[maybe_unused]] is only available in 17 or greater. I don't think it makes much sense to not set it at all.

Comment (ZeroMemes) [reply CMakeLists.txt]: Then it should be set to 17

**Pull request merged**